### PR TITLE
Fixed the logic that prevents all cluster slot requests sent to specific instances

### PR DESCRIPTION
**Pull request by petpetg**
Fixed the logic that prevents all cluster slot requests sent to specific instances when jedis is initialized on many processes(such as storm with a lot of works).
It can cause high CPU and the impact of many slow queries.

Please review.
thanks.

### CODE REVIEW

**Comment by petpetg** [comment]
When the amount of requests is large, it will result in high CPU and a lot of slow queries.

**Comment by petpetg** [comment]
When a storm application with 200 workers is started, the CPU utilization of one instance exceeds 20%, while that of the other instances is only 2%. At the same time, there will be a lot of requests blocked and slow queries.

**Comment by petpetg** [comment]
Hi @sazzad16 
    This changed was fixed the logic that prevents all cluster slot requests sent to specific instances when jedis is initialized on many processes(such as storm with a lot of works). It can cause high CPU and the impact of many slow queries.
    When a storm application with 200 workers is started, the CPU utilization of one instance exceeds 20%, while that of the other instances is only 2%. At the same time, there will be a lot of requests blocked and slow queries.

Please review.
thanks.

**Comment by sazzad16** [comment]
@petpetg I understand and I already reviewed. Please address what was said in the review. Thx!

**Comment by petpetg** [comment]
@sazzad16  Done. Thx!

**Review by dengliming — DISMISSED** [review]

File: src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package redis.clients.jedis;
 
 import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java:6]:
```suggestion
```

File: src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java [hunk 2]
```diff
@@ -117,8 +120,10 @@ public Jedis getConnectionFromNode(HostAndPort node) {
   }
 
   private void initializeSlotsCache(Set<HostAndPort> startNodes, JedisClientConfig clientConfig) {
+    List<HostAndPort> startNodeList = new ArrayList<>(startNodes);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java:123]:
```suggestion
    ArrayList<HostAndPort> startNodeList = new ArrayList<>(startNodes);
```

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**